ts/components/SelectLogoutOption.tsx

```diff
@@ -26,44 +40,51 @@ const styles = StyleSheet.create({
   }
 });
 
-type Props = {
-  onCancel: () => void;
-  onOptionSelected: (_: LogoutOption) => void;
-  header?: React.ReactNode;
-};
-
 /**
  * A modal that allow the user to select the logout method of choice
  */
-class SelectLogoutOption extends React.PureComponent<Props> {
+class SelectLogoutOption extends React.PureComponent<Props, State> {
+  constructor(props: Props) {
+    super(props);
+    this.state = {
+      isLoading: false
+    };
+  }
+
   private onBackPress = () => {
     this.props.onCancel();
     // Returning true is mandatory to avoid the default press action to be
     // triggered as if the modal was not visible
     return true;
   };
 
+  private logout = (logoutOption: LogoutOption) => {
+    this.setState(
+      {
+        isLoading: true
+      },
+      () => this.props.logout(logoutOption)
+    );
+  };
+
   public render() {
-    return (
+    // To avoid reimplementing the Loading component I used the HOC. To be edited with the story: https://www.pivotaltracker.com/story/show/169185930
```

Comment:
```suggestion
    // Using the loading spinner HOC to avoid reimplementing the Loading component
    // TODO - this part can be improved within this story https://www.pivotaltracker.com/story/show/169185930
```